main.go

```diff
@@ -33,60 +39,263 @@ var (
 	cfg = packet.Config{}
 )
 
+type options struct {
+	export            bool
+	armor             bool
+	key               string
+	detachedSignature bool
+	sign              bool
+	input             string
+	output            string
+	name              string
+	comment           string
+	email             string
+}
+
 func main() {
-	if err := run(); err != nil {
-		fmt.Fprintf(os.Stderr, "error: %v", err)
+	var options options
+
+	// common options
+	pflag.BoolVarP(&options.armor, "armor", "a", options.armor, "output in ascii armor")
+	pflag.StringVarP(&options.output, "output", "o", options.output, "write output to file (use - for stdout)")
+
+	// export options
+	pflag.BoolVar(&options.export, "export", options.export, "export public key")
+	pflag.StringVar(&options.name, "name", options.name, "name associated with the key")
+	pflag.StringVar(&options.comment, "comment", options.comment, "comment associated with the key")
+	pflag.StringVar(&options.email, "email", options.email, "email associated with the key")
+
+	// sign options
+	pflag.BoolVarP(&options.sign, "sign", "s", options.sign, "sign a message")
+	pflag.StringVarP(&options.key, "local-user", "u", options.key, "name of key to sign with")
+	pflag.BoolVarP(&options.detachedSignature, "detach-sign", "b", options.detachedSignature, "make a detached signature")
+
+	pflag.CommandLine.ParseErrorsWhitelist.UnknownFlags = true
+	pflag.Parse()
+
+	var err error
+
+	switch {
+	case options.export:
+		args := pflag.Args()
+		if len(args) != 1 {
+			usage()
+		}
+
+		options.key = args[0]
+		err = runExport(options)
+	case options.sign:
+		args := pflag.Args()
+		if len(args) != 1 {
+			usage()
+		}
+
+		options.input = args[0]
+		err = runSign(options)
+	default:
+		usage()
+	}
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+		os.Exit(1)
 	}
 }
 
 func usage() {
-	fmt.Fprintf(os.Stderr, `
-usage: %s generate|sign
-
-Commands:
-	generate:
-		Generate a PGP-compatible public key from the Google Cloud KMS key.
-
-	sign:
-		Sign the input from stdin and produce an ASCII-armored signature on stdout
-
-Environment variables:
-	KEY_NAME:
-		Google Cloud KMS key version resource ID, for example:
-		'projects/$PROJECT/locations/$LOCATION/keyRings/$KEYRING/cryptoKeys/$KEYNAME/cryptoKeyVersions/1'
-	GOOGLE_APPLICATION_CREDENTIALS (optional):
-		Path to Google Cloud credentials file.
-	PGP_UID_NAME (generate only):
-		Name for generated PGP key (e.g., "Phil Zimmermann").
-	PGP_UID_COMMENT (generate only, optional):
-		Comment for generated PGP key.
-	PGP_UID_EMAIL (generate only):
-		Email for generated PGP key (e.g., "phil@example.com")
-`, os.Args[0])
+	pflag.Usage()
```

Comment:
Maybe we could add a little extra usage blurb beyond what pflag returns here? I couldn't figure how to use `--export` correctly. Maybe add a message like `func usage(msg string) {` for clarity?

Reply:
Sure